cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationKeyStoreExtractor.java

```diff
@@ -259,6 +263,24 @@ private static KeyStore retrieveExistingKeyStore(
         }
     }
 
+    @Nonnull
+    private static KeyStore createNewKeyStoreFromPem( @Nonnull final String data, @Nullable final String password )
+    {
+        try {
+            final String s = new String(Base64.getDecoder().decode(data), StandardCharsets.UTF_8);
+            final int del = s.indexOf("-----BEGIN CERTIFICATE-----");
```

Comment:
```suggestion
            final int delimiter = s.indexOf("-----BEGIN CERTIFICATE-----");
```

(minor) some variable names could probably be improved a bit.

Also I am wondering: This currently assumes the key always comes first, right? is this guaranteed to be the case, or could this potentially be an implementation detail of the destination service? If in doubt maybe we just check explicitly?

Reply:
_> This currently assumes the key always comes first, right?_

Yes.

_> is this guaranteed to be the case, or could this potentially be an implementation detail of the destination service?_

Absolutely implementation detail. AFAIK, PEM is completely unordered.

_> If in doubt maybe we just check explicitly?_

Yeah.. would be better, but I was wondering how much effort to expect for enabling PEM support.

This is why I would prefer more integration tests, i.e. manually create PEM files with different orders, with fewer entries, with(out) private-key, etc.